spec/integration/cypress/commands.js

```diff
@@ -0,0 +1,33 @@
+// TODO: Cypress does not support request based whitelisting :( https://github.com/cypress-io/cypress/issues/4843
+const mockRequest = (method, url, fn) => ({
+	method,
+	url,
+	onResponse: (xhr) => {
+		const mockResponse = fn(xhr);
+		if (mockResponse) {
+			xhr.response.body = mockResponse;
+		}
+	}
+});
+
+Cypress.Commands.overwrite("visit", (originalFn, url) => {
+	cy.server();
+	cy.route("POST", "/1/indexes/steplib_inputs/**").as("steplib-inputs");
+
+	cy.fixture("steps.json").then(steps => {
+		cy.route(mockRequest("POST", "/1/indexes/steplib_steps/**", (xhr) => {
+			const { body } = xhr.request;
+			if (body.query === "") {
+				return steps;
+			}
+		}))
```

Comment:
replacing response, when Cypress supports this use case we can migrate to that. (AFAIK it's in progress)